python/cudf/cudf/tests/private_objects/test_memory_records_report.py

```diff
@@ -27,16 +22,39 @@ def rmm_reset():
 
 
 def test_memory_profiling(rmm_reset):
-    df1 = cudf.DataFrame({"a": [1, 2, 3]})
-    assert len(get_memory_records()) == 0
+    import subprocess
+    import sys
+
+    test_code = """
```

Comment:
Please add a comment indicating why we need a subprocess for this test now that it's based on import-time determination of whether nsys is active.